packages/x-charts/src/LineChart/MarkPlot.tsx

```diff
@@ -41,6 +42,12 @@ export interface MarkPlotProps
     event: React.MouseEvent<SVGElement, MouseEvent>,
     lineItemIdentifier: LineItemIdentifier,
   ) => void;
+  /**
+   * If `true` the mark element will only be able to render circle.
+   * Giving less customization options, but saving around 40ms per 1.000 marks.
+   * @default false
+   */
+  perfUpdate?: boolean;
```

Comment:
Why have this prop at all? Couldn't we detect that the shape is a circle and use the more performant circle instead?

Reply:
To avoid breaking changes. This modification implies

- replacing `path` by `circle`
- removing `styled()` which implies not having the theme override

Reply:
All experiments results are available [here](https://www.notion.so/mui-org/charts-Performance-room-for-improvement-8936f96011fb401aaff992374da31efd) in, the section "Using circle instead of path". For removing `styled` we win 6ms

My idea for v8 is to remove those `styled` at this level:
- keep a `styled` at top level (LinePlot, MarkPlot, ChartsXAxis, ...)
- target sub components with classes

For me there is no way to avoid breaking change because `path` -> `circle` is likely to be used for customization. But it's an opportunity to experiment/discuss the future interface

Reply:
Btw, did you try using refs for the shape? Would it work? 🤔

Reply:
What do you mean by "using refs" ?

Reply:
```html
  <circle id="myCircle" cx="5" cy="5" r="4" stroke="blue" />
  <use href="#myCircle" x="10" fill="blue" />
```

Reply:
It does not bring improvement. For a reason I ignore there is no `animated.use` I had to creat by myself

```
const Use = (props) => <use {...props} />

const AnimatedUse = animated(Use)
```